File: backend/src/main/java/com/commercehub/backend/address/service/AddressService.java
package com.commercehub.backend.address.service;

import com.commercehub.backend.address.dto.request.AddressRequest;
import com.commercehub.backend.address.dto.response.AddressResponse;

import java.util.List;

public interface AddressService {

    /**
     * Add a new address for the logged-in user.
     */
    AddressResponse createAddress(AddressRequest request);


    /**
     * Get all addresses of the logged-in user.
     */
    List<AddressResponse> getAllAddresses();


    /**
     * Get address by ID.
     */
    AddressResponse getAddressById(Long addressId);


    /**
     * Update an existing address.
     */
    AddressResponse updateAddress(
            Long addressId,
            AddressRequest request
    );


    /**
     * Delete (soft delete) an address.
     */
    void deleteAddress(Long addressId);


    /**
     * Set an address as default address.
     */
    AddressResponse setDefaultAddress(Long addressId);


    /**
     * Get user's default address.
     */
    AddressResponse getDefaultAddress();

}

File: frontend/src/services/address.service.ts
import apiClient from '../core/api/axios';
import { API_ENDPOINTS } from '../core/api/apiEndpoints';

export const addressService = {
  getAddresses: () => apiClient.get(API_ENDPOINTS.ADDRESSES.BASE),

  createAddress: (payload: Record<string, any>) =>
    apiClient.post(API_ENDPOINTS.ADDRESSES.BASE, payload),

  getAddressById: (addressId: string) =>
    apiClient.get(API_ENDPOINTS.ADDRESSES.BY_ID(addressId)),

  updateAddress: (addressId: string, payload: Record<string, any>) =>
    apiClient.put(API_ENDPOINTS.ADDRESSES.BY_ID(addressId), payload),

  deleteAddress: (addressId: string) =>
    apiClient.delete(API_ENDPOINTS.ADDRESSES.BY_ID(addressId)),

  getDefaultAddress: () => apiClient.get(API_ENDPOINTS.ADDRESSES.DEFAULT),

  setDefaultAddress: (addressId: string) =>
    apiClient.put(API_ENDPOINTS.ADDRESSES.SET_DEFAULT(addressId)),
};
